src/_pytest/assertion/rewrite.py

```diff
@@ -21,7 +22,7 @@
 from _pytest.pathlib import PurePath
 
 # pytest caches rewritten pycs in __pycache__.
-PYTEST_TAG = "{}-PYTEST".format(sys.implementation.cache_tag)
+PYTEST_TAG = "{}-pytest-{}".format(sys.implementation.cache_tag, version)
```

Comment:
Decided to use lowercase `pytest` here because all the other tag elements are lower case. Decided to follow suit.